src/main/java/gregtech/api/capability/impl/miner/MultiblockMinerLogic.java

```diff
@@ -0,0 +1,148 @@
+package gregtech.api.capability.impl.miner;
+
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.recipes.RecipeMap;
+import gregtech.api.render.ICubeRenderer;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.WorldServer;
+import net.minecraft.world.chunk.Chunk;
+
+public class MultiblockMinerLogic extends MinerLogic {
+
+    private static final int CHUNK_LENGTH = 16;
+
+    private final RecipeMap<?> blockDropRecipeMap;
+
+    private int voltageTier;
+    private int overclockAmount = 0;
+
+    private boolean isChunkMode;
+    private boolean isSilkTouchMode;
+
+    /**
+     * Creates the logic for multiblock ore block miners
+     *
+     * @param metaTileEntity the {@link MetaTileEntity} this logic belongs to
+     * @param fortune        the fortune amount to apply when mining ores
+     * @param speed          the speed in ticks per block mined
+     * @param maximumRadius  the maximum radius (square shaped) the miner can mine in
+     */
+    public MultiblockMinerLogic(MetaTileEntity metaTileEntity, int fortune, int speed, int maximumRadius, ICubeRenderer pipeTexture, RecipeMap<?> blockDropRecipeMap) {
+        super(metaTileEntity, fortune, speed, maximumRadius, pipeTexture);
+        this.blockDropRecipeMap = blockDropRecipeMap;
+        this.setActive(false);
+    }
+
+    @Override
+    protected boolean checkShouldStop() {
+        return super.checkShouldStop() || !miner.drainFluid(true);
+    }
```

Comment:
If my suggested changes are made to `checkShouldStop` in `MinerLogic`, this override is not needed anymore.